### PR TITLE
Add lancedb for embeddings

### PR DESCRIPTION
**Pull request by dayesouza**

### CODE REVIEW

File: python/AI/vector_store.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        self.db_connection = lancedb.connect(path)
+        self.db_index = db_index
+        if table_name is not None:
+            self.table = self.db_connection.create_table(
+                table_name, schema=schema, exist_ok=True
+            )
+            self.duckdb_data = self.table.to_lance()
+
+        if self.table is not None and db_index is not None:
+            self.table.create_fts_index(db_index, replace=True)
```
Comment (ha2trinh) [inline python/AI/vector_store.py:35]: would fts index be slower than vector index?

Comment (dayesouza) [reply python/AI/vector_store.py:35]: Actually I'm not using the index at all anymore so I'll remove it for clarity

File: python/AI/embedder.py [hunk 2]
```diff
@@ -4,74 +4,95 @@
...
+schema = pa.schema(
+    [
+        pa.field("hash", pa.string()),
+        pa.field("text", pa.string()),
+        pa.field("vector", pa.list_(pa.float64())),
+    ]
+)
+
 
 class Embedder:
```
[the remark this answers is not in the document]
Comment (dayesouza) [reply python/AI/embedder.py:29]: Yes, great idea! Will do

[the remark this answers is not in the document]
Comment (ha2trinh) [reply python/AI/embedder.py]: changes looked great, thanks!

**Review by ha2trinh — DISMISSED** [review]

**Pull request merged**